src/sentry/tasks/post_process.py

```diff
@@ -293,7 +293,13 @@ def post_process_group(
                         if org_has_commit and features.has(
                             "organizations:workflow-owners", event.project.organization,
                         ):
-                            process_suspect_commits(event=event)
+                            process_suspect_commits.delay(
```

Comment:
The lock a few lines up should be moved into the task instead. It only operates on the group level, and nothing we do out here is group specific.